### PR TITLE
Emmet support for Codeeditor

### PR DESCRIPTION
**Pull request by gpasztor87**

### CODE REVIEW

**Comment by daftspunk** [comment]
Can you provide some documentation on how this is to be used?

**Comment by gpasztor87** [comment]
The documentation and demo is available from here:
http://emmet.io
http://docs.emmet.io

And it was an earlier issue with that: https://github.com/octobercms/october/issues/625

**Comment by isidrorg** [comment]
I agree!! It will be great!!

**Comment by daftspunk** [comment]
Time for me to learn emmet!

Thanks =)

File: modules/backend/formwidgets/codeeditor/assets/vendor/ace/ext-emmet.js [hunk 1]
```diff
@@ -1141,7 +1141,7 @@ var onChangeMode = function(e, target) {
     if (!editor)
         return;
     var modeId = editor.session.$modeId;
-    var enabled = modeId && /css|less|scss|sass|stylus|html|php/.test(modeId);
+    var enabled = modeId && /css|less|scss|sass|stylus|twig|html|php/.test(modeId);
```
Comment (daftspunk) [inline modules/backend/formwidgets/codeeditor/assets/vendor/ace/ext-emmet.js]: It's a shame about this, when we update ACE this could break.

Comment (gpasztor87) [inline modules/backend/formwidgets/codeeditor/assets/vendor/ace/ext-emmet.js]: I don't think so:
https://github.com/ajaxorg/ace-builds/blob/master/src/ext-emmet.js#L1149

Comment (daftspunk) [inline modules/backend/formwidgets/codeeditor/assets/vendor/ace/ext-emmet.js]: Yay!

**Pull request merged**